### DISCUSSION TITLE
Old AdonisJS Legacy 4.1 project broke with npm install

### DISCUSSION THREAD

**Discussion by viniciusmorgado** [adonisjs/core#3990] [Help] [answered]
Hi there!

I'm working with a code base written with the AdonisJS legacy using JavaScript, [version 4.1](https://legacy.adonisjs.com/docs/4.1/about), the problem is that my project always broke when I run npm install and I need to copy the old backup of node_modules from the beginning of the project, the follow error occur:

![WhatsApp Image 2022-11-28 at 20 09 45](https://user-images.githubusercontent.com/34577818/204401096-eecf7f33-ba9b-4183-b7b0-c04f89ce7259.jpeg)

When I look for which modules it is unable to download from npm, I get packages from the framework itself, so the reason for this error is because my project is using a legacy version that is not available in the npm repositories anymore?

If that's the case, is there any way for it to work with npm install without the need to restore the manual node_modules backup? Or is it time to move the project to the latest version of Adonis? I know that it's supports (legacy version) ended in 2021.

Thanks!

**Comment by viniciusmorgado** [comment]
I forget to share my package.json, maybe can be helpful, I remove some info for privacy of the company but in general is very similar:

```
{
  "name": "adonis-api-app",
  "version": "4.1.0",
  "adonis-version": "4.1.0",
  "description": "Adonisjs boilerplate for API server with pre-configured JWT",
  "main": "index.js",
  "scripts": {
    "start": "node server.js",
    "api": "nodemon --inspect server.js",
    "test": "node ace test"
  },
  "keywords": [
    "adonisjs",
    "adonis-app"
  ],
  "author": "",
  "private": true,
  "dependencies": {
    "@adonisjs/ace": "^11.3.1",
    "@adonisjs/antl": "^2.0.7",
    "@adonisjs/auth": "^8.2.3",
    "@adonisjs/bodyparser": "^8.1.7",
    "@adonisjs/cli": "^4.0.13",
    "@adonisjs/cors": "^1.0.7",
    "@adonisjs/fold": "^8.2.0",
    "@adonisjs/framework": "^5.0.13",
    "@adonisjs/ignitor": "^2.0.8",
    "@adonisjs/lucid": "^18.2.0",
    "@adonisjs/lucid-slugify": "^2.2.1",
    "@adonisjs/mail": "^8.1.2",
    "@adonisjs/redis": "^7.3.2",
    "@adonisjs/validator": "^12.4.1",
    "@elastic/elasticsearch": "^8.5.0",
    "@tenda.digital/iugu": "^1.0.12",
    "adonis-acl": "^1.1.1",
    "adonis-throttle": "^2.3.6",
    "array-to-tree": "^3.3.2",
    "axios": "^1.1.3",
    "bugsnag": "^2.4.3",
    "ip": "^1.1.8",
    "iugu": "0.0.6",
    "mysql": "^2.18.1",
    "net-snmp": "^3.8.4",
    "node-virustotal": "^3.35.0",
    "pg": "^8.8.0",
    "random-string": "^0.2.0",
    "slugify": "^1.6.5",
    "uuid": "^9.0.0"
  },
  "devDependencies": {
    "nodemon": "^2.0.20",
    "pm2": "^5.2.2"
  },
  "autoload": {
    "App": "./app"
  }
}
```

**Comment by RomainLanz** [comment, marked as answer]
Hey @viniciusmorgado! 👋🏻 

You have mixed the version of your packages. Some are for V5 (like `Lucid` 18), but you are still using V4.
I recommend downgrading each package to its V4 version or, even better, upgrading to V5.
